src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -591,24 +591,39 @@ drawTPS :: AppState -> Widget Name
 drawTPS s = hBox (tpsInfo : rateInfo)
  where
   tpsInfo
-    | l >= 0 = hBox [str (show n), txt " ", txt (number n "tick"), txt " / s"]
+    | l >= 0 = hBox [str (show n), txt " ", tpsIndicator s, txt (number n "tick"), txt " / s"]
+    -- | l >= 0 = hBox [str (show n), txt " ", txt (number n "tick"), txt " / s"]
     | otherwise = hBox [txt "1 tick / ", str (show n), txt " s"]
 
   rateInfo
     | s ^. uiState . uiGameplay . uiTiming . uiShowFPS =
         [ txt " ("
-        , let tpf = s ^. uiState . uiGameplay . uiTiming . uiTPF
-           in (if tpf >= fromIntegral ticksPerFrameCap then withAttr redAttr else id)
+        , (if tpf >= fromIntegral ticksPerFrameCap then withAttr redAttr else id)
                 (str (printf "%0.1f" tpf))
         , txt " tpf, "
         , str (printf "%0.1f" (s ^. uiState . uiGameplay . uiTiming . uiFPS))
         , txt " fps)"
         ]
     | otherwise = []
 
+  tpf = s ^. uiState . uiGameplay . uiTiming . uiTPF
   l = s ^. uiState . uiGameplay . uiTiming . lgTicksPerSecond
   n = 2 ^ abs l
 
+tpsIndicator :: AppState -> Widget Name
+tpsIndicator s
+  = hBox [txt "(", str $ show tps, txt ") "]
+  -- = str (show $ round tps )
+
+  -- | (tps / target) < 0.9 = str (show $ round tps )
+  -- | otherwise = emptyWidget
+  where
+    tpf = s ^. uiState . uiGameplay . uiTiming . uiTPF
+    fps = s ^. uiState . uiGameplay . uiTiming . uiFPS
+    tps = fps / tpf
```

Comment:
Shouldn't `tps = tpf * fps`?  ticks/frame * frames/sec = ticks/sec.  `fps / tpf` would be in units of square frames per tick-second or something like that.

Reply:
oops.. I guess I didn't do my math right